From f6cd7ea19c4e29f34d15f9650db07a02f7b13f2c Mon Sep 17 00:00:00 2001
From: Pascal Quantin <pascal.quantin@gmail.com>
Date: Tue, 16 Aug 2016 23:52:03 +0200
Subject: [PATCH 128/131] DCT2000: fix a buffer overflow in case of malformed
 packet

Bug: 12750
Change-Id: Iff20efa46e000ab4ae77b2edbcb0b9af4634508c
Reviewed-on: https://code.wireshark.org/review/17096
Reviewed-by: Pascal Quantin <pascal.quantin@gmail.com>
Petri-Dish: Pascal Quantin <pascal.quantin@gmail.com>
Tested-by: Petri Dish Buildbot <buildbot-no-reply@wireshark.org>
Reviewed-by: Anders Broman <a.broman58@gmail.com>
(cherry picked from commit 2e37b271c473e1cbd01d62ebe1f3b011fc9fe638)
Reviewed-on: https://code.wireshark.org/review/17106
(cherry picked from commit 2024fd0427a372e6e1d05f980d05430ef9c8c187)
Reviewed-on: https://code.wireshark.org/review/17798
Reviewed-by: Balint Reczey <balint@balintreczey.hu>
---
 epan/dissectors/packet-catapult-dct2000.c | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/epan/dissectors/packet-catapult-dct2000.c b/epan/dissectors/packet-catapult-dct2000.c
index 2fa25b7..4cef69c 100644
--- a/epan/dissectors/packet-catapult-dct2000.c
+++ b/epan/dissectors/packet-catapult-dct2000.c
@@ -1587,6 +1587,9 @@ static void attach_fp_info(packet_info *pinfo, gboolean received, const char *pr
 
     /* Number of channels (for coordinated channels) */
     p_fp_info->num_chans = outhdr_values[i++];
+    if (p_fp_info->num_chans > MAX_FP_CHANS) {
+        p_fp_info->num_chans = MAX_FP_CHANS;
+    }
 
     /* EDCH-Common is always T2 */
     if (p_fp_info->channel == CHANNEL_EDCH_COMMON) {
-- 
2.1.4

